### PR TITLE
Add missing typescript type definitions

### PR DESCRIPTION
**Pull request by programmer24601**
This PR closes issue #659

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1,7 +1,203 @@
...
+    dirtyHeight: number,
+  ): void
+  getImageData(sx: number, sy: number, sw: number, sh: number): ImageData
+  createImageData(sw: number, sh: number): ImageData
+  createImageData(imagedata: ImageData): ImageData
+  /**
+   * For PDF canvases, adds another page. If width and/or height are omitted,
+   * the canvas's initial size is used.
+   */
+  addPage(width?: number, height?: number): void
```
Comment (twlite) [inline index.d.ts:87]: PDF is not supported in this library though 🤔

Comment (twlite) [reply index.d.ts:87]: #309

Comment (programmer24601) [reply index.d.ts:87]: Apologies I should have clarified this.

The following types were directly copied from `skia-canvas` which this project depends on:
```
GlobalCompositeOperation
CanvasLineCap
CanvasLineJoin
CanvasTextBaseline
CanvasTextAlign
TextMetrics
CanvasFillRule
CanvasRenderingContext2D
CanvasGradient
CanvasPattern
```
It is correct to be bringing in these types, but as you rightfully point out, this library does not currently implement all features. I was reluctant to start removing type properties / class methods as I'm uncertain which are, and are not, implemented. I'm hoping someone with more experience (i.e. the maintainers) can help out here. 

The only modifications I've made are to `CanvasRenderingContext2D`:
1. To add method `setTransform(a: number, b: number, c: number, d: number, e: number, f: number): void` in addition to the existing `setTransform(transform?: DOMMatrix2D): void`
2. To modify `fill(fillRule?: CanvasFillRule): void` to `fill(path?: Path2D, fillRule?: CanvasFillRule): void`

These changes now allow my project to build (`tsc`) where before it was failing (as described in #659).

**Comment by programmer24601** [comment]
PR #659 has been resolved another way.

**Pull request closed without merging**